src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -2067,28 +2102,66 @@ public void NewPeerClaimsHeadSecondTime_PeerDisconnected_ChainShouldNotChange()
             var chainedHeadersWithPeerK = new Dictionary<uint256, ChainedHeader>(cht.GetChainedHeadersByHash());
 
             cht.PeerDisconnected(4);
-
+            
             Dictionary<uint256, ChainedHeader> chainedHeadersWithoutPeerK = cht.GetChainedHeadersByHash();
 
             chainedHeadersWithoutPeerK.Should().BeEquivalentTo(chainedHeadersWithPeerK);
 
-            this.CheckChainedHeaderTreeConsistency(cht, ctx, consensusTip, new HashSet<int>() { 0, 1, 2, 3 });
+            this.CheckChainedHeaderTreeConsistency(cht, ctx, consensusTip);
+        }
+
+        /// <summary>
+        /// Issue 20 @ New peer K is connected, it prolongs D by 2 headers. K is disconnected, only those 2 headers are removed.
```

Comment:
Issue 20 @ New peer K is connected, it prolongs it prolongs D's chain by 2 headers. K is disconnected, only those 2 headers are removed.